### ISSUE TITLE
DevOps: add SonarQube to the repo for code QC

### ISSUE THREAD

**Issue by nlebovits** [CodeForPhilly/clean-and-green-philly#751]
## Describe the task
We would like to integrate [the community edition of SonarQube](https://www.sonarsource.com/open-source-editions/sonarqube-community-edition/) into the repository with GitHub actions. The goal is to improve our devops and reduce the load on our leads when reviewing PRs.

## Acceptance Criteria

- [ ] SonarQube configuration files added to the repository
- [ ] SonarQube GitHub Actions Workflow added to the repository for both the Python and JavaScript/TypeScript parts of the codebase
- [ ] Relevant secrets stored in the GitHub repo (contact @CodeWritingCow or @nlebovits if you run into permissions issues)
- [ ] Add SonarQube configurations to the `.vscode/settings.json` to improve dev experience in VSCode

**Comment by csharpie** [comment]
Hello, I'd like to work on this issue. Please assign it to me.

**Comment by nlebovits** [comment]
@csharpie Assigned!

**Comment by csharpie** [comment]
Thanks Nissim.

Brad

**Comment by csharpie** [comment]
I'll add a note on Slack too but it looks like SonarQube's Community Edition (because it's installed locally) doesn't support Pull Request Decorations. However there's an unofficial GitHub that does support <a href='https://github.com/mc1arke/sonarqube-community-branch-plugin' target='_blank'>this</a>. I plan to look into this to see the feasibility.

**Comment by CodeWritingCow** [comment]
Hello @csharpie, any update on this ticket? We normally reassign the ticket if there's no activity on it for 1 week. I wanted to reach out before doing that. Thanks!

**Comment by csharpie** [comment]
Hey Gary,

I spent some time on it yesterday and made significant progress I think. I
will update the issue today and let you know where I’ve left off.

Thanks for following up.

Sincerely,
Brad Steinberg

**Comment by csharpie** [comment]
Hi Team,

I unfortunately hit a bit of roadblock with getting the yml file built for this and I unfortunately don't have a working GitHub workflow yml file to commit. Here's what I've learned so far in hopes that this helps whomever takes this issue on next.

SonarQube is not available in a clouded hosted Community Edition but SonarCloud is and is available at the Free Tier for Open Source Projects! I'd highly recommend reading through this <a href="https://dev.to/jei/part-2-automating-code-quality-scanning-using-sonar-cloud-and-github-actions-8g7" target="_blank">Dev.to article</a>. These are articles and documentation that I found helpful for getting me up to speed on SonarCloud with GitHub Actions:

- https://dev.to/remast/go-for-sonarcloud-with-github-actions-3pmn
- https://medium.com/@rahulsharan512/integrating-sonarcloud-with-github-actions-for-secure-code-analysis-26a7fa206d40
- https://docs.sonarsource.com/sonarqube/latest/devops-platform-integration/github-integration/adding-analysis-to-github-actions-workflow/
